### PR TITLE
Flp 950 remove durable entities

### PR DESCRIPTION
**Pull request by MichaelYoung1981**

### CODE REVIEW

File: src/Command/FreezePayments/FreezePaymentsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿namespace SFA.DAS.Funding.ApprenticeshipPayments.Command.FreezePayments
+{
```
Comment (MikeV-TalentConsulting) [inline src/Command/FreezePayments/FreezePaymentsCommand.cs:2]: most of the files use file scoped namespaces

File: src/Command/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -18,6 +20,8 @@ public static IServiceCollection AddCommandServices(this IServiceCollection serv
             serviceCollection.AddScoped<IProcessUnfundedPaymentsCommandHandler, ProcessUnfundedPaymentsCommandHandler>();
             serviceCollection.AddScoped<ICalculateRequiredLevyAmountCommandHandler, CalculateRequiredLevyAmountCommandHandler>();
             serviceCollection.AddScoped<IRecalculateApprenticeshipPaymentsCommandHandler, RecalculateApprenticeshipPaymentsCommandHandler>();
+            serviceCollection.AddScoped<IFreezePaymentsCommandHandler, FreezePaymentsCommandHandler>();
```
Comment (MikeV-TalentConsulting) [inline src/Command/ServiceCollectionExtensions.cs:23]: Inconsistent indentation

File: src/DataAccess/SFA.DAS.Funding.ApprenticeshipPayments.DataAccess.csproj [hunk 3]
```diff
@@ -0,0 +1,28 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net6.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <ItemGroup>
+	  <PackageReference Include="Azure.Identity" Version="1.12.0" />
```
Comment (MikeV-TalentConsulting) [inline src/DataAccess/SFA.DAS.Funding.ApprenticeshipPayments.DataAccess.csproj:10]: Indentation

File: src/Command/ProcessUnfundedPayments/ProcessUnfundedPaymentsCommandHandler.cs [hunk 4]
```diff
@@ -25,79 +23,39 @@ public ProcessUnfundedPaymentsCommandHandler(
...
         {
-            _logger.LogInformation("Apprenticeship Key: {apprenticeshipKey} -  Publishing {numberOfPayments} payments for collection period {collectionPeriod} & year {collectionYear}", apprenticeshipKey, paymentsToSend.Length, command.CollectionPeriod, command.CollectionYear);
+            _logger.LogInformation("Apprenticeship Key: {apprenticeshipKey} -  Publishing {numberOfPayments} payments for collection period {collectionPeriod} & year {collectionYear}", apprenticeshipKey, paymentsToSend.Count, command.CollectionPeriod, command.CollectionYear);
         }
         else
         {
             _logger.LogInformation("Apprenticeship Key: {apprenticeshipKey} -  No payments to publish for collection period {collectionPeriod} & year {collectionYear}", apprenticeshipKey, command.CollectionPeriod, command.CollectionYear);
         }
 
+        apprenticeship.MarkPaymentsAsSent(command.CollectionYear, command.CollectionPeriod);
```
Comment (MikeV-TalentConsulting) [inline src/Command/ProcessUnfundedPayments/ProcessUnfundedPaymentsCommandHandler.cs:48]: There is a danger here that a payment is marked as sent, but then fails to send in the foreach loop, would be better to mark individual payments as they are sent.

File: src/Domain/Apprenticeship/Payment.cs [hunk 5]
```diff
@@ -14,13 +21,34 @@ public Payment(short academicYear, byte deliveryPeriod, decimal amount, short co
...
+        }
+
+        public void Unfreeze(short collectionYear, byte collectionPeriod)
+        {
+            NotPaidDueToFreeze = false;
+            CollectionYear = collectionYear;
+            CollectionPeriod = collectionPeriod;
+        }
+
+        public void Send()
```
Comment (MikeV-TalentConsulting) [inline src/Domain/Apprenticeship/Payment.cs:49]: Should probably rename this to MarkAsSent

File: src/Infrastructure/Configuration/ApplicationSettings.cs [hunk 6]
```diff
@@ -10,6 +10,7 @@ public class ApplicationSettings
     public string NServiceBusConnectionString { get; set; }
     public string DCServiceBusConnectionString { get; set; }
     public string NServiceBusLicense { get; set; }
+    public string DbConnectionString { get; set; }
```
Comment (MikeV-TalentConsulting) [inline src/Infrastructure/Configuration/ApplicationSettings.cs:13]: I can't see a PR for the config change related to this, is it done already?

File: src/Command/ServiceCollectionExtensions.cs (hunk 2 above)
Comment (tom-gough) [reply src/Command/ServiceCollectionExtensions.cs:23]: this looks fine to me? both here and in vs

File: src/DurableEntities/PaymentsFunctions.cs [hunk 7]
```diff
@@ -1,34 +1,43 @@
...
+		var previousAcademicYear = await GetPreviousAcademicYear();
+		
+        var result = await _getApprenticeshipsWithDuePaymentsQueryHandler.Get(new GetApprenticeshipsWithDuePaymentsQuery(releasePaymentsCommand.CollectionPeriod, releasePaymentsCommand.CollectionYear));
 
-        var previousAcademicYear = await GetPreviousAcademicYear();
+        var releasePaymentsTasks = result.Apprenticeships.Select(x => _processUnfundedPaymentsCommandHandler.Process(new ProcessUnfundedPaymentsCommand(releasePaymentsCommand.CollectionPeriod, releasePaymentsCommand.CollectionYear, x.ApprenticeshipKey, short.Parse(previousAcademicYear.AcademicYear), previousAcademicYear.HardCloseDate)));
+        log.LogInformation($"Releasing payments for collection period {releasePaymentsCommand.CollectionPeriod} & year {releasePaymentsCommand.CollectionYear} for apprenticeships. (Count: {result.Apprenticeships.Count()})");
+        await Task.WhenAll(releasePaymentsTasks);
 
         var operationInput = new ReleasePaymentsDto
```
Comment (tom-gough) [inline src/DurableEntities/PaymentsFunctions.cs:42]: Is this object needed anymore?

File: src/Command/ServiceCollectionExtensions.cs (hunk 2 above)
Comment (tom-gough) [reply src/Command/ServiceCollectionExtensions.cs:23]: this looks ok to me in both vs and github

File: src/DataAccess/ImplicitUsings.cs [hunk 8]
```diff
@@ -0,0 +1 @@
+﻿
```
Comment (tom-gough) [inline src/DataAccess/ImplicitUsings.cs:1]: Can we remove this if we are not adding any usings into here?

**Pull request closed without merging**